Ruins/src/main/java/atomicstryker/ruins/common/RuinTemplate.java

```diff
@@ -1248,4 +1310,34 @@ public RuinTemplateRule getVariant(final int selectorInitial, final int ruleInde
             }
         }
     }
+
+    public static class ClassException extends RuntimeException
+    {
+        private static final long serialVersionUID = -7630819126112904714L;
+
+        protected ClassException(String message)

```

Comment:
ClassException is a bad name, TemplateLoadingException maybe? Also, adding three classes for 2 errors which are actually somewhat the same is over-engineering things, IMO

Reply:
I want something a little more specific, as callers may want to distinguish between this particular situation and something else. Other exceptions may indicate more serious issues and warrant corrective action or urgent log messages; this one is pretty benign and could very well be intentional/expected. I went with IncompatibleModException.